src/test/java/uk/ac/ebi/ampt2d/accession/file/FileAccessioningDatabaseServiceTest.java

```diff
@@ -0,0 +1,96 @@
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.file;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.accession.file.persistence.FileAccessioningDatabaseService;
+import uk.ac.ebi.ampt2d.accession.file.rest.FileDTO;
+import uk.ac.ebi.ampt2d.commons.accession.generators.ModelHashAccession;
+import uk.ac.ebi.ampt2d.test.configurationaccession.FileAccessioningServiceTestConfiguration;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@TestPropertySource(properties = "services=file-accession")
+@Import(FileAccessioningServiceTestConfiguration.class)
+public class FileAccessioningDatabaseServiceTest {
+
+    private static final String CHECKSUM_A = "checksumA";
+    private static final String CHECKSUM_B = "checksumB";
+
+    @Autowired
+    private FileAccessioningDatabaseService databaseService;
+
+    @Test
+    public void testFileAccessionsAreStoredInTheRepositoryAndRetrived() throws Exception {
+        List<ModelHashAccession<FileModel, String, String>> fileAccessionsToStore = Arrays.asList(
+                ModelHashAccession.of((FileModel) new FileDTO(CHECKSUM_A), CHECKSUM_A, CHECKSUM_A),
+                ModelHashAccession.of((FileModel) new FileDTO(CHECKSUM_B), CHECKSUM_B, CHECKSUM_B)
+        );
+
+        databaseService.save(fileAccessionsToStore);
+
+        Collection<String> messageHashes = Arrays.asList(CHECKSUM_A, CHECKSUM_B);
+        Map<String, ? extends FileModel> accessionsFromRepository =
+                databaseService.findAllAccessionsByHash(messageHashes);
+
+        assertTrue(accessionsFromRepository.containsKey(CHECKSUM_A));
+        assertTrue(accessionsFromRepository.containsKey(CHECKSUM_B));
+    }
+
+    @Test
+    public void addingTheSameFilesWithSameAccessionsTwiceOverwritesObject() throws Exception {
+        List<ModelHashAccession<FileModel, String, String>> fileAccessionsToStore = Arrays.asList(
+                ModelHashAccession.of((FileModel) new FileDTO(CHECKSUM_A), CHECKSUM_A, CHECKSUM_A),
+                ModelHashAccession.of((FileModel) new FileDTO(CHECKSUM_B), CHECKSUM_B, CHECKSUM_B)
+        );
+
+        databaseService.save(fileAccessionsToStore);
+        databaseService.save(fileAccessionsToStore);
+
+        Collection<String> messageHashes = Arrays.asList(CHECKSUM_A, CHECKSUM_B);
+        Map<String, ? extends FileModel> accessionsFromRepository =
+                databaseService.findAllAccessionsByHash(messageHashes);
+
+        assertTrue(accessionsFromRepository.containsKey(CHECKSUM_A));
```

Comment:
We are checking here that the two objects were written, but we are not sure if there are more objects in the repository, so we are not sure if they were overwritten or not.

Reply:
@DataJpaTest -> all tests are performed inside independant transactions and rolls back any database change after finishing unless specified on a test.